libs/math/functions.h

```diff
@@ -152,6 +158,42 @@ sinc (T const& x)
     return (x == T(0) ? T(1) : std::sin(x) / x);
 }
 
+/* ----------------------------- Popcount ------------------------- */
+
+/** Returns the number of one bits of an integer. */
+template <typename T>
+std::size_t constexpr
+popcount (T const x)
+{
+    static_assert(std::is_integral<T>::value && std::is_unsigned<T>::value,
+        "T must be an unsigned integral type.");
+    return std::bitset<sizeof(T) * 8>(x).count();
+}
+
+#ifdef _MSC_VER // MSVS fails to optimize std::bitset::count
+std::size_t inline
+popcount (std::uint32_t const x)
+{
+#   if !(__amd64__ || __x86_64__ || _WIN64 || _M_X64)
+    __pragma (message("Warning: CPU support for _mm_popcnt_u32 is assumed, "
+        "but could not be verified."))
+#   endif
```

Comment:
Not sure how to check for CPU support here and settled on just assuming availability and showing a warning for now.